src/edu/washington/escience/myria/operator/SymmetricHashJoin.java

```diff
@@ -331,7 +331,13 @@ public SymmetricHashJoin(final List<String> outputColumns, final Operator left,
   @Override
   protected Schema generateSchema() {
     final Schema leftSchema = getLeft().getSchema();
+    if (leftSchema == null) {
```

Comment:
No, it's okay for `generateSchema()` to return `null`, e.g. https://github.com/uwescience/myria/blob/master/src/edu/washington/escience/myria/operator/Apply.java#L124. I modified `SymmetricHashJoin` only because of the potential `NullPointerException`. I haven't noticed any other operator with the same problem.